cli/core/validator.go

```diff
@@ -2,17 +2,91 @@ package core
 
 import (
 	"context"
+	"fmt"
 	"math/big"
 	"strconv"
 
 	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
 	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/fatih/color"
 )
 
-func GenerateValidatorProof(ctx context.Context, eigenpodAddress string, eth *ethclient.Client, chainId *big.Int, beaconClient BeaconClient) (*eigenpodproofs.VerifyValidatorFieldsCallParams, []uint64) {
+func SubmitValidatorProof(ctx context.Context, owner, eigenpodAddress string, chainId *big.Int, eth *ethclient.Client, batchSize uint64, proofs *eigenpodproofs.VerifyValidatorFieldsCallParams, noPrompt bool) ([]*types.Transaction, error) {
+	ownerAccount, err := PrepareAccount(&owner, chainId)
+	if err != nil {
+		return nil, err
+	}
+	PanicOnError("failed to parse private key", err)
+
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+	if err != nil {
+		return nil, err
+	}
+
+	indices := Uint64ArrayToBigIntArray(proofs.ValidatorIndices)
+	validatorIndicesChunks := chunk(indices, batchSize)
+	validatorProofsChunks := chunk(proofs.ValidatorFieldsProofs, batchSize)
+	validatorFieldsChunks := chunk(proofs.ValidatorFields, batchSize)
+	if !noPrompt {
+		PanicIfNoConsent(fmt.Sprintf("This will call EigenPod.VerifyWithdrawalCredentials() %d times, to link your validator to your eigenpod.", len(validatorIndicesChunks)))
```

Comment:
Could be improved by adding estimated gas cost and number of validators this will link

Reply:
yep, next iteration